lib/factories/ppt/index.js

```diff
@@ -15,12 +16,14 @@ let { TableStylesFactory } = require('./table-styles');
 let { ThemeFactory } = require('./theme');
 let { ViewPropsFactory } = require('./view-props');
 let { MediaFactory } = require('./media');
+let { EmbeddingsFactory } = require('./embeddings');
+let { ChartFactory } = require('./charts');
 let { PptFactoryHelper } = require('../../helpers/ppt-factory-helper');
 
 class PptFactory {
-    constructor(presentation, args) {
-        this.content = presentation.content;
-        this.presentation = presentation;
+    constructor(parentFactory, args) {
+        this.content = parentFactory.content;
```

Comment:
It's a ref to an object - that's how I designed it. Other factories can write their respective content, and in the end, the highest level factory will automatically have the correct content stemming from the root. Note however, some of the factories don't "section off" their part of the content, but this is something I've been meaning to change for a while, just haven't got around to it yet.

Reply:
ok yeah i worry about object pollution of the state if multiple factories have access to the same object but can't rely on a given state. It may lead to hard to track bugs at some point.